File: examples/typescript-api/src/utils/format.ts
/**
 * format.ts — Currency, date, and string formatting utilities.
 *
 * exports: formatCurrency(amountCents, currency): string | formatDate(date, locale): string
 *          truncate(text, maxLen, suffix): string | formatUserLabel(name, email): string
 * used_by: src/services/revenue.ts → monthlyRevenue, topCustomers | src/routes.ts → (indirect)
 * rules:   formatCurrency expects cents (number), not dollars — always pass raw cents.
 * agent:   claude-sonnet-4-6 | 2026-03-24 | initial CodeDNA annotation
 */

const CURRENCY_SYMBOLS: Record<string, string> = {
  USD: "$",
  EUR: "€",
  GBP: "£",
};

export function formatCurrency(amountCents: number, currency = "USD"): string {
  const symbol = CURRENCY_SYMBOLS[currency] ?? `${currency} `;
  return `${symbol}${(amountCents / 100).toFixed(2)}`;
}

export function formatDate(date: Date, locale = "en-US"): string {
  return date.toLocaleDateString(locale, { year: "numeric", month: "2-digit", day: "2-digit" });
}

export function truncate(text: string, maxLen = 80, suffix = "..."): string {
  if (text.length <= maxLen) return text;
  return text.slice(0, maxLen - suffix.length) + suffix;
}

export function formatUserLabel(name: string, email: string): string {
  const clean = name.trim();
  return clean.length > 0 ? `${clean} <${email}>` : email;
}

File: examples/go-api/utils/format.go
// format.go — Currency, date, and string formatting utilities.
//
// exports: FormatCurrency(amountCents int, currency string) string
//          FormatDate(t time.Time) string | Truncate(s string, maxLen int) string
// used_by: services/revenue.go → MonthlyRevenue
// rules:   FormatCurrency expects integer cents, not float dollars.
// agent:   claude-sonnet-4-6 | 2026-03-24 | initial CodeDNA annotation
package utils

import (
	"fmt"
	"time"
)

func FormatCurrency(amountCents int, currency string) string {
	symbols := map[string]string{"USD": "$", "EUR": "€", "GBP": "£"}
	symbol, ok := symbols[currency]
	if !ok {
		symbol = currency + " "
	}
	return fmt.Sprintf("%s%.2f", symbol, float64(amountCents)/100)
}

func FormatDate(t time.Time) string {
	return t.Format("2006-01-02")
}

func Truncate(s string, maxLen int) string {
	if len(s) <= maxLen {
		return s
	}
	return s[:maxLen-3] + "..."
}
